zaza/openstack/charm_tests/rabbitmq_server/tests.py

```diff
@@ -0,0 +1,406 @@
+        rmq_utils.wait_for_cluster()
+
+        # Negative check - tcp connect to management plugin port
+        logging.info('Expect tcp connect fail since charm config '
+                     'option is disabled.')
+        tries = 0
+        ret = generic_utils.port_knock_units(units,
+                                             mgmt_port,
+                                             expect_success=False)
+
+        while ret and tries < (max_wait / 30):
+            time.sleep(30)
+            logging.debug('Attempt {}: {}'.format(tries, ret))
+            ret = generic_utils.port_knock_units(units, mgmt_port,
+                                                 expect_success=False)
+            tries += 1
+
+        self.assertIsNone(ret)
+        logging.info('Confirm mgmt port closed on all units (OK)\n')
+
+    def test_414_rmq_nrpe_monitors(self):
+        """Check rabbimq-server nrpe monitor basic functionality."""
+        units = zaza.model.get_units(self.application_name)
+        host_names = generic_utils.get_unit_hostnames(units)
+
+        # check_rabbitmq monitor
+        logging.debug('Checking nrpe check_rabbitmq on units...')
+        cmds = ['egrep -oh /usr/local.* /etc/nagios/nrpe.d/'
+                'check_rabbitmq.cfg']
+        ret = generic_utils.check_commands_on_units(cmds, units)
+        self.assertIsNone(ret)
+
+        logging.debug('Sleeping 2ms for 1m cron job to run...')
+        time.sleep(120)
+
+        # check_rabbitmq_queue monitor
+        logging.debug('Checking nrpe check_rabbitmq_queue on units...')
+        cmds = ['egrep -oh /usr/local.* /etc/nagios/nrpe.d/'
+                'check_rabbitmq_queue.cfg']
+        ret = generic_utils.check_commands_on_units(cmds, units)
+        self.assertIsNone(ret)
+
+        # check dat file existence
+        logging.debug('Checking nrpe dat file existence on units...')
+        for u in units:
+            unit_host_name = host_names[u.entity_id]
+
+            cmds = [
+                'stat /var/lib/rabbitmq/data/{}_general_stats.dat'.format(
+                    unit_host_name),
+                'stat /var/lib/rabbitmq/data/{}_queue_stats.dat'.format(
+                    unit_host_name)
+            ]
+
+            ret = generic_utils.check_commands_on_units(cmds, [u])
+            self.assertIsNone(ret)
+
+        logging.info('OK\n')
+
+    def test_910_pause_and_resume(self):
+        """The services can be paused and resumed."""
+        logging.debug('Checking pause and resume actions...')
+
+        unit = zaza.model.get_units(self.application_name)[0]
+        assert unit.workload_status == "active"
+
+        zaza.model.run_action(unit.entity_id, "pause")
+        zaza.model.block_until_unit_wl_status(unit.entity_id, "maintenance")
+        # TODO: investigate possible bug (the following line is
+        # required, otherwise it looks like workload_status is
+        # reporting cached information, no matter how long you sleep)
```

Comment:
This makes sense as we are working asynchronously. The version of the unit object you have until now has a status of "active". The recheck is required.